seedpod_ground_risk/core/plot_server.py

```diff
@@ -337,14 +358,23 @@ def generate_path_data_popup(self, layer):
         from seedpod_ground_risk.pathfinding.environment import GridEnvironment
         from seedpod_ground_risk.ui_resources.info_popups import DataWindow
         from seedpod_ground_risk.layers.fatality_risk_layer import FatalityRiskLayer
-        for i in self.data_layers:
-            if isinstance(i, FatalityRiskLayer):
-                cur_layer = GridEnvironment(self._generated_data_layers['Fatality Risk'][1])
+        for dlayer in self.data_layers:
+            if isinstance(dlayer, FatalityRiskLayer) and layer.aircraft == dlayer.ac_dict:
+                cur_layer = GridEnvironment(self._generated_data_layers[dlayer.key][1])
                 grid = cur_layer.grid
                 popup = DataWindow(layer, grid)
                 popup.exec()
                 break
 
+    def remove_duplicate_layers(self):
```

Comment:
This should be replaced by overriding the builtin `__eq__` method of the class and have that compare the `ac_dict` and `bounds` or similar